### ISSUE TITLE
Why is kernel_weight of nvdla_compiler just a part of weight?

### ISSUE THREAD

**Issue by MINZHIJI** [nvdla/sw#47]
I trace `wisdon.dir` which was generated by `nvdla_compiler`.
And I found the folder named `kernel_weights` has a file named `values`. I use hexdump to read `value` file.
I found the compiler will translate caffemodel weight to `value` file in `kernel_weight` folder.

**Then, I have some questions ...!** ([ Data Link](https://drive.google.com/open?id=1XFZSDp688OqWDmk7bJjb8NNfsZwnTpyO) - I provide my observed resource.)
1. **Why Why is kernel_weight just a part of total weight?**
For example, conv1 of Lenet is 5 x 5 x 20 kernel, so the weight maybe have 500 fp32 values.
Then, the `conv1/kernel_weight/value` need to have 500 values.
But I read `value` by "Float - Mid-Little Endian", and the `value` just has 125 fp32 values!!

**`value` hexdump result**
```
22c4 3e80 d100 3e5c f7b0 3e05 cafe 3e6c
3331 3e06 6052 be4b fff0 3e38 947c be0e
7598 bde0 6e92 3bf0 334c beb3 bc20 bdb1
7b4f 3e88 f644 3e1c ae52 3c0b 74aa be5b
309d 3e8f 38e8 3e8d 314c be1a 85e5 bcea
9f7f 3e92 32d0 3e20 2965 bd5f 73bc be8f
9cc9 bbcd d9a6 3c9a ef7e bcaf f218 3ea7
f01b 3e8e 753d bde9 596a 3a97 85ae bd87
e028 3e4d fe14 be2a 61d5 be68 464c be00
73f3 be81 85e2 be13 7421 bd80 8ea3 3e0d
8c1e beb0 9552 be24 7de3 be95 174a 3e23
432e 3d9b 1442 be83 eb07 3dbf 3742 be7b
8060 3dac 4b7b bdb3 7b8e be94 8225 be54
9e96 3dd8 f076 3e9f 09c4 bda4 fae2 bea9
90ce bea2 3765 3e3a 68a4 3e98 ddd1 3ec6
fb5d bedd 39c9 bd88 9719 3f08 6676 3ec4
a0eb 3d84 66ef be2d 861f 3e85 ea16 3ebb
912f 3e94 ac51 be5b af91 3cca 89cc 3eb8
e86d 3ea5 034b bdb5 fa1e be9d 598a 3db9
c4ee be5d 55e1 3c0d 20ff 3eb7 ac5e 3e4e
3e7e bb90 2b23 bb44 0560 3ebf d6ad 3e57
aabf 3e9f 60c7 bdd3 f17a 3eaa c0b0 3e40
c08a bd9b 1712 bded 2756 3e44 d999 3e11
fc75 be29 14f4 be93 58e8 be9f 722a bdf5
0ad1 bd74 339b be9c 7eba 3cfd b80d 3c00
c4db 3e50 a08b 3e64 9fe6 3e4d 4fdf be78
ced5 be83 0de4 be02 67d6 bb6e 7032 3eb9
bf23 be88 e583 be61 7af3 3e8e 9832 3e46
6bb2 3e91 9852 3e09 579e bea6 ce63 3e05
48fe 3e82 0587 be19 18a4 bcb7 13a2 3e18
aa83 3ea6 52bc 3e3b 25ba ba95 c70a be93
caf9 bd99                              
```
**Use Float32 - Mid-Little Endian to translate `value`**
I use the website to translate. ([Link](https://www.scadacore.com/tools/programming-calculators/online-hex-converter/))
![image](https://user-images.githubusercontent.com/37866456/40352816-77042b3e-5de2-11e8-9b29-561a3060b139.png)
![image](https://user-images.githubusercontent.com/37866456/40353150-5c98a030-5de3-11e8-97dd-a7b7a448fc71.png)

**The caffemodel translate to fp32**
```
000  2.5026524e-01 
001  2.1564102e-01 
002  1.3082767e-01 
003  2.3124310e-01 
004  1.3105465e-01 
005 -1.9860962e-01 
 ...
125 -7.5094171e-02 
 ...
```
The last value of `value` is `caf9 bd99` , and its fp32 is `-0.07509417`
**Then, where are the rest weights after 125th?**

2. **NVDLA `loadable` file weight values are int8, int16 or fp16? And what is the relationship `loadable` and  `wisdom.dir`?**

**Comment by JunningWu** [comment]
in order to run cifar10_quick on nv_small, we need to generate int8 caffemodel /loadable file. 
do you know which one, nvdla_compiler or caffe, to finish this?

**Comment by MINZHIJI** [comment]
@JunningWu  Currently, I don't have any int8 inference model. 
And Can we management to generated the different precision of weights with nvdla compiler?

**Comment by JunningWu** [comment]
@MINZHIJI have you find why the compiled wisdon.dir's values less than the caffemodel's?

**Comment by MINZHIJI** [comment]
@JunningWu  Nope. I have not known this whether related with final generated file. So I think that continuing to trace is nonsense!  :(

**Comment by wyxsky** [comment]
I tried to print the value of caffe layers' weight, and files in wisdon.dir, found it's just a quarter of the weights of each layer, do not have any compression, so I guess it's useless information or temporary files for nvdla_compiler.
